src/Common/tests/System/Collections/IEnumerable.Generic.Tests.cs

```diff
@@ -169,27 +175,35 @@ protected enum EnumerableOrder
         }
 
         private void VerifyEnumerator(
-            IEnumerator enumerator,
+            IEnumerator<T> enumerator,
             T[] expectedItems,
             int startIndex,
             int count,
             bool validateStart,
             bool validateEnd)
         {
-            bool needToMatchAllExpectedItems = count - startIndex
-                                               == expectedItems.Length;
+            bool needToMatchAllExpectedItems = count - startIndex == expectedItems.Length;
             if (validateStart)
             {
                 for (var i = 0; i < 3; i++)
                 {
-                    if (IEnumerable_Generic_Enumerator_Current_EnumerationNotStarted_ThrowsInvalidOperationException)
+                    if (Enumerator_Generic_Current_UndefinedOperation_Throws)
                     {
                         Assert.Throws<InvalidOperationException>(() => enumerator.Current);
                     }
                     else
                     {
                         var cur = enumerator.Current;
                     }
+
+                    if (Enumerator_NonGeneric_Current_UndefinedOperation_Throws)
```

Comment:
Tests for the non-generic IEnumerable can be in the non-generic counterpart of this class, which is just for the generic IEnumerable.  And it looks like tests like this already exist there, at least in part.